tests/transforms/test_sign_expand.py

```diff
@@ -0,0 +1,194 @@
+            qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match=r"Passed tape must end in"):
+            tapes, fn = qml.transforms.sign_expand(tape)
+
+    def test_hamiltonian_error_not_jointly_measurable(self):
+        with pennylane.tape.QuantumTape() as tape:
+            H_mult = 1.5 * qml.PauliZ(0) + 2 * qml.PauliZ(1) + 0.3 * qml.PauliX(0)
+            qml.expval(H_mult)
+
+        with pytest.raises(ValueError, match=r"Passed hamiltonian"):
+            tapes, fn = qml.transforms.sign_expand(tape)
+
+    @pytest.mark.parametrize(("tape", "output"), zip(TAPES_var, OUTPUTS_var))
+    def test_hamiltonians_vars(self, tape, output):
+        """Tests that the hamiltonian_expand transform returns the correct value"""
+
+        tapes, fn = qml.transforms.sign_expand(tape)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert np.isclose(output, expval)
+
+    @pytest.mark.parametrize(("tape", "output"), zip(TAPES_var, OUTPUTS_var))
+    def test_hamiltonians_vars_circuit_impl(self, tape, output):
+        """Tests that the hamiltonian_expand transform returns the correct value
+        if we do not calculate analytical expectation values of groups but rely on their circuit approximations"""
+
+        tapes, fn = qml.transforms.sign_expand(tape, circuit=True)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert np.isclose(output, expval, 1e-1)
+
+    @pytest.mark.autograd
+    def test_hamiltonian_dif_autograd(self, tol):
+        """Tests that the hamiltonian_expand tape transform is differentiable with the Autograd interface"""
+
+        H = qml.Hamiltonian(
+            [-0.2, 0.5, 1], [qml.PauliX(3), qml.PauliZ(1) @ qml.PauliY(2), qml.PauliZ(0)]
+        )
+
+        var = pnp.array([0.1, 0.67, 0.3, 0.4, -0.5, 0.7, -0.2, 0.5, 1.0], requires_grad=True)
+        output = 0.42294409781940356
+
+        with qml.tape.QuantumTape() as tape:
+            for i in range(2):
+                qml.RX(np.array(0), wires=0)
+                qml.RX(np.array(0), wires=1)
+                qml.RX(np.array(0), wires=2)
+                qml.CNOT(wires=[0, 1])
+                qml.CNOT(wires=[1, 2])
+                qml.CNOT(wires=[2, 0])
+            qml.expval(H)
+
+        def cost(x):
+            tape.set_parameters(x, trainable_only=False)
+            tapes, fn = qml.transforms.sign_expand(tape, circuit=True)
+            res = qml.execute(tapes, dev, qml.gradients.param_shift)
+            return fn(res)
+
+        def cost_trad(x):
+            tape.set_parameters(x, trainable_only=False)
+            tapes, fn = qml.transforms.hamiltonian_expand(tape)
+            res = qml.execute(tapes, dev, qml.gradients.param_shift)
+            return fn(res)
+
+        assert np.isclose(cost(var), cost_trad(var), 0.05)
+
+        # TODO: This does not work yet due to qml.utils.sparse_hamiltonian does not allow autograd to push gradients through
```

Comment:
Did you try to use any other gradient strategy? See [here](https://docs.pennylane.ai/en/stable/introduction/interfaces.html?highlight=gradient#gradients)